kernel/syscall.c

```diff
@@ -5712,6 +5734,18 @@ static long _syscall(void* args_)
 long myst_syscall(long n, long params[6])
 {
     long ret;
+
+    /* Userspace can query fs and gs via arch_prctl,
+    handle it here, as _syscall switches thread descriptors */
+    if (n == SYS_arch_prctl)
+    {
+        int code = (int)params[0];
+        unsigned long* addr = (unsigned long*)params[1];
+
+        _strace(n, "code=%d addr=%p\n", code, addr);
```

Comment:
I didn't see the problem with ``_strace`` when we discussed this before. I have a different idea that might work. We can move this to the ``_syscall`` function as an ordinary case of the switch statement, except we add some code to the beginning of ``_syscall`` that "caches" the value of ``fsbase`` and ``gsbase`` at the start of the ``_syscall`` function before the prologue. Then you can just pass these values into ``myst_syscall_arch_prtcl`` and then use them as needed.

Reply:
I didn't realize it either that this would break `_strace`. Have moved the syscall to be handled in the switch table. It is using a cached value for fs. Do we need to cache gs?